src/Workspace/ErrorWorkspacePlaceholder.jsx

```diff
@@ -1,21 +1,33 @@
 import React from 'react';
+import { useHistory } from 'react-router-dom/cjs/react-router-dom';
+import isEnabled from '../helpers/featureFlags';
 import './Workspace.less';
 import NotFoundSVG from '../img/not-found.svg';
 import { components } from '../params';
+import workspaceRegistrationConfig from '../localconf';
 
-class ErrorWorkspacePlaceholder extends React.Component {
-  render() {
-    const supportEmail = components.login?.email || 'support@datacommons.io';
-    return (
-      <div className='error-workspace-placeholder__error-msg'>
-        <h1>Error opening workspace...</h1>
-        <p>
-        Workspace access requires authorization. Please contact <a href={`mailto:${supportEmail}`}>{supportEmail}</a> for more information.
-        </p>
-        <NotFoundSVG />
-      </div>
-    );
+const ErrorWorkspacePlaceholder = () => {
+  const history = useHistory();
+  if (isEnabled('workspaceRegistration')) {
```

Comment:
I think you should wrap these stuff in a useEffect hook so it will be called only once when the component mounts. Otherwise it will be called each time this component renders, which is unnecessary

Reply:
Since we have agreed to change the implementation to use `<Redirect>` instead of `useHistory()`